fibers.scm

```diff
@@ -120,11 +120,21 @@
       (unless init
         (error "run-fibers requires initial fiber thunk when creating sched"))
       (spawn-fiber (lambda ()
-                     (call-with-values init
-                       (lambda vals (atomic-box-set! ret vals)))
-                     ;; Could be that this fiber was migrated away.
-                     ;; Make sure to wake up the main scheduler.
-                     (spawn-fiber (lambda () #t) scheduler))
+                     (dynamic-wind
+		       (const #t)
+		       (lambda ()
+			 (catch #t
+			   (lambda ()
+			     (call-with-values init
+			       (lambda vals (atomic-box-set! ret
+							     (list 'ok vals)))))
```

Comment:
It doesn't make a huge difference, but for a tiny bit extra performance / less memory usage and simplicity, you could just drop the `(list 'ok ...)` prefix.  Basically, just stuff the list in ret on success, and stuff `#false` in case of failure.

Reply:
Nevermind, that's bogus because of exceptions.

Reply:
Actually, if you go for with-exception-handler / raise-style exceptions, then a condition and a list can be distinguished, OTOH in principle it's allowed to raise non-conditions, so maybe let's not.